tests/channel_int/channel_int.c

```diff
@@ -232,19 +234,25 @@ TEST_FUNCTION(test_channel_create_and_destroy)
     /// arrange
 
     /// act
-    THANDLE(CHANNEL) channel = channel_create(g.g_threadpool);
+    THANDLE(THREADPOOL) threadpool = create_threadpool();
+
+    THANDLE(CHANNEL) channel = channel_create(threadpool);
 
     /// assert
     ASSERT_IS_NOT_NULL(channel);
 
     // cleanup
     THANDLE_ASSIGN(CHANNEL)(&channel, NULL);
+    threadpool_close(threadpool);
+    THANDLE_ASSIGN(THREADPOOL)(&threadpool, NULL);
```

Comment:
Isn't the fact that we're creating a threadpool for each test hiding a deeper issue?
I am not getting how this is solving the crash actually ...

Reply:
Fixed, this is also spawned a new task here [https://msazure.visualstudio.com/One/_workitems/edit/25500968](https://msazure.visualstudio.com/One/_workitems/edit/25500968)